.pre-commit-config.yaml

```diff
@@ -1,15 +1,23 @@
 repos:
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.3.4
+    rev: v0.5.3
```

Comment:
It looks like bumping the version here is causing some new hits when running on `Lib/test/`. Maybe we should leave bumping the version (and fixing the newly uncovered bugs) to a followup

Reply:
Great minds... I've just pushed a revert commit, and agree re followup.

Reply:
Passing: https://github.com/python/cpython/actions/runs/10006647644/job/27659751581?pr=122018